frontend/lib/src/components/shared/BaseColorPicker/BaseColorPicker.tsx

```diff
@@ -47,113 +49,121 @@ export interface BaseColorPickerProps {
   help?: string
 }
 
-interface State {
-  /**
-   * The value specified by the user via the UI. If the user didn't touch this
-   * widget's UI, the default value is used.
-   */
-  value: string
-}
-
-class BaseColorPicker extends React.PureComponent<
-  BaseColorPickerProps,
-  State
-> {
-  public state: State = {
-    value: this.props.value,
-  }
+const BaseColorPicker = (props: BaseColorPickerProps): React.ReactElement => {
+  const {
+    width,
+    showValue,
+    label,
+    labelVisibility,
+    help,
+    disabled,
+    value: propValue,
+  } = props
+  const [value, setValue] = React.useState(propValue)
+  const [popoverKey, setPopoverKey] = React.useState(0)
+  const theme: EmotionTheme = useTheme()
 
-  public componentDidUpdate(prevProps: BaseColorPickerProps): void {
-    if (
-      prevProps.value !== this.props.value &&
-      this.props.value !== this.state.value
-    ) {
-      this.setState((_, prevProps) => {
-        return { value: prevProps.value }
-      })
+  React.useEffect(() => {
+    // 2021.06.30 - on Streamlit Sharing, ColorPicker throws a cross-origin
+    // error when its popover window is closed. There's an issue open in the
+    // react-color repo https://github.com/casesandberg/react-color/issues/806 -
+    // but it's months old and hasn't had a developer response.
+    const handleError = (error: ErrorEvent): void => {
+      if (error.error?.name === "SecurityError") {
+        logWarning(
+          `Swallowing ColorPicker SecurityError '${error.error.name}: ${error.error.message}'`
+        )
+        // We force an update by changing the key of the popover after this error,
+        // to re-mount the UIPopover - because the error sometimes cause it to be
+        // unmounted. This is an unfortunate hack.
+        setPopoverKey(prev => prev + 1)
```

Comment:
I *think* this does the same thing as before but I'm not sure if it always works well and if that's the most elegant solution.